<launch>
  <arg name="joy"        default="true" />
  <arg name="joy_dev"    default="$(optenv SPOT_JOY_DEVICE /dev/input/js0)" />
  <arg name="joy_config" default="$(eval optenv('SPOT_JOY_CONFIG', find('spot_driver') + '/config/teleop.yaml'))" />

  <node pkg="interactive_marker_twist_server" type="marker_server" name="twist_marker_server" >
    <rosparam command="load" file="$(find spot_driver)/config/planar.yaml" />
    <param name="link_name" value="body" />
    <param name="robot_name" value="spot" />
  </node>

  <group if="$(arg joy)">
    <rosparam command="load" ns="bluetooth_teleop" file="$(arg joy_config)" />

    <group ns="bluetooth_teleop">
      <param name="joy_node/dev" value="$(arg joy_dev)" />
      <node pkg="joy" type="joy_node" name="joy_node" />
      <!-- <node pkg="teleop_twist_joy" type="teleop_node" name="teleop_twist_joy"/> -->
      <node pkg="spot_driver" type="spot_controller" name="F710_controller" output="screen">
        <remap from="estop/gentle" to="/spot/estop/gentle" />
        <remap from="stand" to="/spot/stand" />
        <remap from="sit" to="/spot/sit" />
        <remap from="stair_mode" to="/spot/stair_mode" />
        <remap from="locomotion_mode" to="/spot/locomotion_mode" />
        <remap from="body_pose" to="/spot/body_pose" />
      </node>
    </group>
  </group>
</launch>
